src/charm.py

```diff
@@ -604,6 +616,31 @@ def _push_to_workload(self, filename, content, event):
             LOGGER.info("workload container not ready - deferring")
             event.defer()
 
+    def _update_trusted_ca_certs(self, container: Container) -> bool:
+        """Update trusted CA certificates with the cert from configuration.
+
+        Livepatch needs to restart to use newly received certificates.
+
+        Args:
+            container (Container): The workload container, the caller must ensure that we can connect.
+
+        Returns:
+            bool: A boolean to indicate whether the workload service should be restarted.
+        """
+        if not self.config.get("contracts.ca"):
+            LOGGER.error("ca config not set")
+            return False
+
+        LOGGER.error("pushing ca cert")
+        cert = b64decode(self.config.get("contracts.ca")).decode("utf8")
+        container.push(TRUSTED_CA_FILENAME, cert, make_dirs=True)
+        LOGGER.error("updating ca cert")
+        stdout, stderr = container.exec(["update-ca-certificates", "--fresh"]).wait_output()
+        LOGGER.info("stdout update-ca-certificates: %s", stdout)
+        LOGGER.info("stderr update-ca-certificates: %s", stderr)
+
+        return True
```

Comment:
One issue here is that we will always force restart if a custom ca is set. What do you think about adding some logic to detect if a new CA was added to the pool? One idea is checking the output from update-ca-certificates.

Reply:
now storing cert hash in _state and updating only if hash changes